iroh-net/src/tls/resolver.rs

```diff
@@ -0,0 +1,87 @@
+use std::sync::Arc;
+
+use iroh_base::key::SecretKey;
+use webpki::types::{pem::PemObject, CertificateDer, PrivatePkcs8KeyDer};
+
+use super::{certificate, CreateConfigError};
+use crate::tls::Authentication;
+
+#[derive(Debug)]
+pub(crate) struct AlwaysResolvesCert {
```

Comment:
Kind of annoying that `rustls` has *all* the implementations for `ResolvesClientCert` and `ResolvesServerCert` that we need, but they're not all exposed.

There's
- `AlwaysResolvesClientCert`
- `AlwaysResolvesServerCert`
- `AlwaysResolvesClientRawPublicKeyCert`
- `AlwaysResolvesServerRawPublicKeyCert`

I'd rather not have to implement a trait outside the crate where it's defined, but eh.
Also it's super annoying that they split it up into client/server like that.